app/controllers/auth_controller.rb

```diff
@@ -1,18 +1,18 @@
 class AuthController < ApplicationController
-  before_filter :setup_mygov_access_token
-
+  before_action :setup_mygov_access_token
 
   def oauth_callback
-    auth = request.env['omniauth.auth']
-    return_to = self.return_to.try(:path)
+    auth = request.env["omniauth.auth"]
 
-    sign_out
-    user = User.from_oauth_hash(auth)
-    sign_in(user)
+    # fetch the redirect path (if any) before the authn sign in/out
```

Comment:
@jessieay good point. I hadn't considered that `return_to` would now persist for **any** request across the login. Test added in 1687353